src/libraries/System.IO.FileSystem/tests/File/Copy.cs

```diff
@@ -350,6 +352,22 @@ public void WindowsAlternateDataStreamOverwrite(string defaultStream, string alt
             Assert.Throws<IOException>(() => Copy(testFileAlternateStream, testFile2, overwrite: true));
             Assert.Throws<IOException>(() => Copy(testFileAlternateStream, testFile2 + alternateStream, overwrite: true));
         }
+
+        [Fact]
+        public void DestinationFileIsTruncatedWhenItsLargerThanSourceFile()
+        {
+            string sourcePath = GetTestFilePath();
+            string destPath = GetTestFilePath();
+
+            const int SourceFileSize = 1000;
+            File.WriteAllBytes(sourcePath, RandomNumberGenerator.GetBytes(SourceFileSize));
+            File.WriteAllBytes(destPath, RandomNumberGenerator.GetBytes(SourceFileSize * 2));
+
+            Copy(sourcePath, destPath, overwrite: true);
+
+            using SafeFileHandle destHandle = File.OpenHandle(destPath, FileMode.Open);
+            Assert.Equal(SourceFileSize, RandomAccess.GetLength(destHandle));
```

Comment:
assert that the contents match? one could imagine a bug that corrupts something.